src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs

```diff
@@ -103,6 +104,22 @@ internal static bool IsNullableType(ITypeSymbol typeOpt)
         {
             return typeOpt?.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
         }
+        
+        internal static bool IsNullableOfBoolean(ITypeSymbol type)
+        {
+            return IsNullableType(type) && IsBooleanType(GetNullableUnderlyingType(type));
+        }
+
+        internal static ITypeSymbol GetNullableUnderlyingType(ITypeSymbol type)
+        {
+            Debug.Assert(IsNullableType(type));
+            return ((INamedTypeSymbol)type).TypeArguments[0];
+        }
+
+        internal static bool IsBooleanType(ITypeSymbol type)
+        {
+            return type.SpecialType == SpecialType.System_Boolean;
```

Comment:
>. [](start = 23, length = 1)

Should we use conditional access here to avoid null checks in callers for error cases?

Reply:
> Should we use conditional access here to avoid null checks in callers for error cases?

So far we didn't have to make the null checks at the call sites. However, in next PRs I'll add conditional access to these helpers to make CFG builder code and future usages more robust. 

---
In reply to: [189610315](https://github.com/dotnet/roslyn/pull/26994#discussion_r189610315) [](ancestors = 189610315)